maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/inclusivenaming/InclusiveNamingRule.java

```diff
@@ -0,0 +1,239 @@
+                .filter(this::include)
+                .forEach(fr -> {
+                    count.incrementAndGet();
+                    processFile(fr, failures);
+                });
+
+        if (failures.size() == 0) {
+            Log.info("Inclusive naming processed " + count.get() + " files.");
+        } else {
+            Log.info("Inclusive naming processed " + count.get() + " files, found " + failures.size() + " errors");
+        }
+        return failures;
+    }
+
+    private void processFile(FileRequest fr, List<RuleFailure> errors) {
+        try (BufferedReader bufferedReader = Files.newBufferedReader(fr.path(), StandardCharsets.UTF_8)) {
+            String line;
+            int lineNum = 0;
+            List<Term> found = new LinkedList<>();
+            while ((line = bufferedReader.readLine()) != null) {
+                // we want to start from 1
+                lineNum++;
+                for (Term term : terms) {
+                    if (line.toLowerCase().contains(term.term)) {
+                        found.add(term);
+                    }
+                }
+                if (!found.isEmpty()) {
+                    errors.add(RuleFailure.create(fr, lineNum, "inclusive naming found: " + String.join(", ", found.toString())));
+                }
+                found.clear();
+            }
+
+        } catch (IOException e) {
+            throw new EnforcerException("Failed to process file for inclusive naming: " + fr.relativePath());
+        }
+    }
+
+    private boolean include(FileRequest fr) {
+        for (FileMatcher exclude : excludes) {
+            if (exclude.matches(fr)) {
+                return false;
+            }
+        }
+
+        for (FileMatcher include : includes) {
+            if (include.matches(fr)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Fluent API builder for {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+     */
+    public static class Builder {
+        private InlusiveNamingConfig inclusiveNamingConfig;
+
+        private Builder() {
+        }
+
+        /**
+         * Update builder from maven configuration.
+         *
+         * @param inclusiveNamingConfig maven config
+         * @return updated builder
+         */
+        public Builder config(InlusiveNamingConfig inclusiveNamingConfig) {
+            this.inclusiveNamingConfig = inclusiveNamingConfig;
+            return this;
+        }
+
+        /**
+         * Build a new instance of {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+         *
+         * @return new rule
+         */
+        public InclusiveNamingRule build() {
+            return new InclusiveNamingRule(this);
+        }
+
+        List<Term> terms() {
+            List<Term> terms = new ArrayList<>();
+            try (InputStream is = InclusiveNamingRule.class.getResourceAsStream("index.json")) {
```

Comment:
Why using JSON here ? Is that file standard ?
Can we use XML instead ? This way we dodge the whole Jakarta thing.
We can use plexus-xml or our custom parser (see common/xml). 

We can also publish a schema at https://helidon.io/xsd.

Reply:
That JSON comes from here: https://inclusivenaming.org/word-lists/index.json

If we allow the file to be set by users as an optional parameter, it is better to keep it exactly in the same format. Otherwise it will be an overhead to them.

Reply:
I don't think anyone else other than us is consuming this plugin. It is odd for a maven plugin to externalize its configuration as JSON.

We will need the built-in words and at least configuration for additional words and exclusions.
Maybe we don't need to expose configuration for an external file at all ?

Reply:
It is not a configuration file, it is a data file, in the same way it is possible to specify the template for the copyright.

There  was a suggestion already about externalize this file, and it is implemented:
https://github.com/helidon-io/helidon-build-tools/pull/837#discussion_r1061597708

Maybe you are right and nobody else will use it, but I think we should not put this constraint ourselves. Additionally, developing this constraint requires more effort. It looks this is not correct to me.

Adding additional words and exclusions can be done, although it goes beyond the task. It will not be easy for the user to configure them in the plugin, because each word contains replacements, explication, etc...

Reply:
Lets imagine we do this in XML. There are some concerns about it.

Scenario 1: We allow to set the inclusive-naming file in XML as a maven parameter
Users will need to know how to transform the JSON into XML. So we need to develop one tool, and this tool needs to be known by them.

Scenario 2: We don't allow to set the inclusive-naming file as a maven parameter.
Users will need to have an updated plugin that will contain an updated file of inclusive0naming. In this case, they have to wait that we release a new version of the plugin.

In both scenarios we will need one tool to parse the XML into JSON, because we don't want to do this by hand. Where should we place this tool?. In both scenarios I also expect users complaining or thinking about why do we make it more complicated than it should be.

Regarding the exclusions, okay, I will implement it.

Reply:
This plugin is primarily intended for Helidon projects ; not for general consumption. We don't care for external users and thus don't need to provide a tool for converting the official JSON file to our own custom XML file.

The conversion should be done once (e.g. using an online converter), it's fine to let it become stale and perform manual updates sporadically.

Remember that this is about performing a check for Helidon projects, not providing a general inclusive naming tool.

Reply:
@tomas-langer you requested in this PR to be able to configure the file in the plugin (this is already in the PR). Do you agree with @romain-grecourt about not allowing that?, and also to use the file internally as an XML?.

Reply:
@romain-grecourt I modified it to work with XML